scripts/visualizer.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+PKG = 'embodied_attention'
+import roslib; roslib.load_manifest(PKG)
+
+import rospy
+from geometry_msgs.msg import Point
+from sensor_msgs.msg import Image
+from std_msgs.msg import Float64
+from cv_bridge import CvBridge, CvBridgeError
+import cv2 as cv
+import sys
+
+class Visualizer():
+    def __init__(self):
+        image_sub = rospy.Subscriber("/icub_model/left_eye_camera/image_raw", Image, self.image_callback, queue_size=1, buff_size=2**24)
+
+        target_sub = rospy.Subscriber("/saccade_target", Point, self.target_callback, queue_size=1, buff_size=2**24)
+        potential_target_sub = rospy.Subscriber("/saccade_potential_target", Point, self.potential_target_callback, queue_size=1, buff_size=2**24)
+
+        pan_sub = rospy.Subscriber("/robot/left_eye_pan/pos", Float64, self.pan_callback, queue_size=1, buff_size=2**24)
+        tilt_sub = rospy.Subscriber("/robot/eye_tilt/pos", Float64, self.tilt_callback, queue_size=1, buff_size=2**24)
+
+        self.image_augmented_pub = rospy.Publisher("/image_visualized", Image, queue_size=1)
+
+        self.image = None
+
+        self.targets = []
+        self.potential_targets = []
+
+        self.saliency_width = float(rospy.get_param('~saliency_width', '256'))
+        self.saliency_height = float(rospy.get_param('~saliency_height', '192'))
+
+        self.cv_bridge = CvBridge()
+
+    def target_callback(self, target):
+        # scale to camera image size
+        x = int(target.x * (float(self.image.width)/self.saliency_width))
+        y = int(target.y * (float(self.image.height)/self.saliency_height))
+
+        self.targets.append((x, y))
+        self.publish()
+
+    def potential_target_callback(self, target):
+        # scale to camera image size
+        x = int(target.x * (float(self.image.width)/self.saliency_width))
+        y = int(target.y * (float(self.image.height)/self.saliency_height))
+
+        self.potential_targets.append((x, y))
+        self.publish()
+
+    def publish(self):
+        if self.image is not None:
+            try:
+                image = self.cv_bridge.imgmsg_to_cv2(self.image, "bgr8")
+            except CvBridgeError as e:
+                print e
+
+            # visualization
+            for t in self.potential_targets:
+                cv.circle(image, (t[0], t[1]), 2, (0, 255, 0))
+            for t in self.targets:
```

Comment:
this is why you have more than one targets per image; instead of a list you should maybe only take the latest message (for potential targets too)